pkg/cmd/install.go

```diff
@@ -86,16 +91,32 @@ func MakeInstall() *cobra.Command {
 
 		fmt.Printf("Result: %s %s\n", string(res.StdOut), string(res.StdErr))
 
-		absPath, _ := filepath.Abs(localKubeconfig)
-		fmt.Printf("Saving file to: %s\n", absPath)
-
 		kubeconfig := strings.Replace(string(res.StdOut), "localhost", ip.String(), -1)
 
-		writeErr := ioutil.WriteFile(absPath, []byte(kubeconfig), 0600)
-		if writeErr != nil {
-			return writeErr
+		if merge {
+			// Create a merged kubeconfig
+			data, err := mergeConfigs(localKubeconfig, []byte(kubeconfig))
+			if err != nil {
+				return err
+			}
+			if writeErr := writeConfig(localKubeconfig, data); writeErr != nil {
```

Comment:
writeConfig is duplicated now, we will only need it once as long as we didn't return due to an error

Reply:
@alexellis I am not able to get what you're trying to tell me. If I'm right you're telling me to use writeConfig only once? But there are two separate instances where we want to use it,
- One when we want to merge and the mergeConfigs() function returns a slice of data bytes to be passed into writeConfig()
- Another instance of using writeConfig() is when the user just wants to create a new kubeconfig.

Since the returned slice of data in the first case if scoped inside of the if. writeConfig() needs to be duplicated for the two instances. Any suggestions?

Reply:
This is the section of code I'm asking about
`kubeconfig := strings.Replace(string(res.StdOut), "localhost", ip.String(), -1)`

`if merge {`
`        // Create a merged kubeconfig`
`	data, err := mergeConfigs(localKubeconfig, []byte(kubeconfig))`
`	if err != nil {`
`		return err`
`	}`
`	if writeErr := writeConfig(localKubeconfig, []byte(data)); writeErr != nil {`
`		return writeErr`
`	}`
`	} else {`
`		// Create a new kubeconfig`
`		if writeErr := writeConfig(localKubeconfig, []byte(kubeconfig)); writeErr != nil {`
`			return writeErr`
`		}`
`        }`

Reply:
So you'd use something like

```go
var localKubeConfig string

if x {
 localKubeConfig = "etc etc"
} else {
 localKubeConfig = "etc etc"
}

writeToDisk(localKubeConfig)
```

In that way you avoid the duplication.

Reply:
Thanks a lot! Didn't know the solution would be this simple 😅

Reply:
Done! 😄